### PR TITLE
[Settings] Hide "Push Rules" option in Advanced settings>Notifications #271

### PR DESCRIPTION
**Pull request by ClaireGizard**
#271 

Screenshot_1638975597

### CODE REVIEW

**Review by giomfo — APPROVED** [review]
SGTM

File: vector/src/main/res/xml/vector_settings_advanced_settings.xml [hunk 1]
```diff
@@ -73,10 +73,10 @@
...
             app:fragment="im.vector.app.features.settings.push.PushGatewaysFragment" />
 
-        <im.vector.app.core.preference.VectorPreference
-            android:persistent="false"
-            android:title="@string/settings_push_rules"
-            app:fragment="im.vector.app.features.settings.push.PushRulesFragment" />
+<!--        <im.vector.app.core.preference.VectorPreference-->
+<!--            android:persistent="false"-->
+<!--            android:title="@string/settings_push_rules"-->
+<!--            app:fragment="im.vector.app.features.settings.push.PushRulesFragment" />-->
```
Comment (Florian14) [inline vector/src/main/res/xml/vector_settings_advanced_settings.xml:79]: For other Tchap hidden settings, we set the `isPreferenceVisible` field to false

File: vector/src/main/res/xml/vector_settings_advanced_settings.xml [hunk 2]
```diff
@@ -76,7 +76,8 @@
         <im.vector.app.core.preference.VectorPreference
             android:persistent="false"
             android:title="@string/settings_push_rules"
-            app:fragment="im.vector.app.features.settings.push.PushRulesFragment" />
+            app:fragment="im.vector.app.features.settings.push.PushRulesFragment"
+            app:isPreferenceVisible="@bool/false_tchap_hidden"/>
```
Comment (Florian14) [inline vector/src/main/res/xml/vector_settings_advanced_settings.xml:80]: Please reformat, a space is missing before `/>`

**Review by Florian14 — APPROVED** [review]
LGTM

**Pull request merged**